javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -449,6 +446,29 @@ public final void write(ClientModel model, JavaFile javaFile) {
         });
     }
 
+    /**
+     * Override parent setters if:
+     * 1. parent property is not readOnly or required
+     * 2. child does not contain property that shadow this parent property, otherwise overridden parent setter methods will collide with child setter methods
+     * @see <a href="https://github.com/Azure/autorest.java/issues/1320">Issue 1320</a>
+     */
+    protected List<ClientModelPropertyAccess> getParentSettersToOverride(ClientModel model, JavaSettings settings, List<ClientModelPropertyReference> propertyReferences) {
+        return propertyReferences.stream()
+            .filter(ClientModelPropertyReference::isFromParentModel)
+            .map(ClientModelPropertyReference::getReferenceProperty)
+            .filter(parentProperty -> {
+                // parent property is not readOnly or required
+                if (parentProperty.getIsReadOnly() ||
+                    (settings.isRequiredFieldsAsConstructorArgs() && parentProperty.isRequired())) {
+                    return false;
+                }
+                // child does not contain property that shadow this parent property
+                Set<String> modelProperties = model.getProperties().stream().map(ClientModelProperty::getName).collect(Collectors.toSet());
```

Comment:
Could this be put to top of the method? Seems no need to run every lambda.

Reply:
Edited.